pandas/tests/frame/test_indexing.py

```diff
@@ -2939,6 +2939,15 @@ def test_where_callable(self):
         tm.assert_frame_equal(result,
                               (df + 2).where((df + 2) > 8, (df + 2) + 10))
 
+    def test_where_empty_df_and_empty_cond_having_non_bool_dtypes(self):
+        # GH 21947
+        data = []
+        df = pd.DataFrame(columns=['a'], data=data)
+        cond = df.applymap(lambda x: x > 0)
+        # Should not raise ValueError:
+        # Boolean array expected for the condition, not object
+        df.where(cond)
```

Comment:
can you check the result and assert e.g.

```
result = df.where(cond)
expected = ....
tm.assert_frame_equal(...)
```

Reply:
Modification done. 
Even though I initially wrote the test this way on purpose, as I only want to test if an error is raised.